Client/Clique/Modules/CreateEvent/BasicInfo/BasicInfoViewController.swift

```diff
@@ -0,0 +1,213 @@
+    var pickerViewCategory = UIPickerView()
+    var pickerViewCurrency = UIPickerView()
+    var category = ""
+    var currency = ""
+    var cost = 0
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Basic info (1/5)"
+        costSwitcher.addTarget(self, action: #selector(costInputEnabled(switcher:)), for: .valueChanged)
+
+        categoryTextField.delegate = self
+        currencyTextField.delegate = self
+        costTextField.delegate = self
+        pickerViewCategory.delegate = self
+        pickerViewCategory.dataSource = self
+        pickerViewCurrency.delegate = self
+        pickerViewCurrency.dataSource = self
+        categoryTextField.inputView = pickerViewCategory
+        categoryTextField.tintColor = UIColor.clear
+        currencyTextField.inputView = pickerViewCurrency
+        currencyTextField.tintColor = UIColor.clear
+        costTextField.isHidden = true
+        currencyTextField.isHidden = true
+        downButtonCurrency.isHidden = true
+        amountLabel.isHidden = true
+        costTextField.text = "0"
+        
+        getCategories()
+        getCurrencies()
+    }
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        guard
+            var selectedCategory = categoryTextField.text,
+            var eventName = nameTextField.text,
+            var participantNumber = participantsTextField.text,
+            !selectedCategory.isEmpty && !eventName.isEmpty && !participantNumber.isEmpty
+        else{
+            alert(fwdMessage: "Please fill all required information.")
+            return
+        }
+        if segue.identifier == "sendInfo" {
+            let controller = segue.destination as! DateTimeViewController
+            controller.selectedCategory = categoryTextField.text!
+            controller.eventName = nameTextField.text!
+            controller.participantNumber = participantsTextField.text!
+            controller.eventCost = costTextField.text!
+            controller.currency = currencyTextField.text!
+            }
+        
+    }
+    @IBAction func downButtonCategoryPressed(_ sender: UIButton) {
+        categoryTextField.becomeFirstResponder()
+    }
+    
+    @IBAction func downButtonCyrrencyPressed(_ sender: UIButton) {
+        currencyTextField.becomeFirstResponder()
+    }
+    
+    @IBAction func nextButtonPressed(_ sender: UIBarButtonItem) {
+        guard let viewContoller = UIStoryboard(name: "DateTime", bundle: nil).instantiateInitialViewController() as? DateTimeViewController else{
+            return
+        }
+        navigationController?.pushViewController(viewContoller, animated: true)
+    }
+    
+    func getCategories() {
+        createEventService.getCategories{ result in
+            switch result {
+            case .success(let categories):
+                for category in categories{
+                    self.categories.append(category.category_name)
+                }
```

Comment:
you need to use weak self, also use map here, something like:
categories = categories.map(\.categoryName)